### PR TITLE
feat(vscode): adds new-middleware command

### PR DESCRIPTION
**Pull request by alestiago**
## Status

**READY**

## Description

Adds the ability to create middleware from the context menu or command palette.

## Demonstration

https://github.com/VeryGoodOpenSource/dart_frog/assets/44524995/a4d68e38-6c9e-4ff8-b949-6b6c627f9bae

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: extensions/vscode/src/commands/new-middleware.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+ * When launching the command from the context menu, the Uri corresponds to the
+ * selected file or directory. Only those directories or dart files under a
+ * `routes` directory show the command in the context menu. Therefore the user
+ * does not need to select a directory or file via the open dialog.
+ *
+ * All the logic associated with creating a new route is handled by the
+ * `dart_frog new middleware` command, from the Dart Frog CLI.
+ *
+ * @see [Dart Frog CLI `new` command implementation](https://github.com/VeryGoodOpenSource/dart_frog/tree/main/packages/dart_frog_cli/lib/src/commands/new)
+ * @param { Uri | undefined} uri
```
Comment (erickzanardo) [inline extensions/vscode/src/commands/new-middleware.ts:24]:
```suggestion
 * @param {Uri | undefined} uri
```
or
```suggestion
 * @param { Uri | undefined } uri
```

**Review by erickzanardo — COMMENTED** [review]
I don't remember 100% of the code of the create command, but I feel that this new command a very similar code to it, I wonder if it wouldn't be possible to reuse some of that code.

**Comment by alestiago** [comment]
> https://github.com/VeryGoodOpenSource/dart_frog/pull/727#pullrequestreview-1498954905

Yes, that's a good idea. I do plan to introduce a separate refactoring pull request to reuse the code.

**Review by erickzanardo — APPROVED** [review]
I feel there may be changes overall depending on how the discussion on https://github.com/VeryGoodOpenSource/dart_frog/issues/701 goes.

But so far, code LGTM and any changes that may or may not arise from the discussion, can be addressed in different PRs.

**Comment by alestiago** [comment]
Merging this in 😄 💙 !

As suggested by Eric, any further changes can be addressed in a different Pull Request.

**Pull request merged**